runtime/gc_structs/ClassLoaderClassesIterator.cpp

```diff
@@ -108,9 +123,13 @@ GC_ClassLoaderClassesIterator::nextClass()
 		if (ANONYMOUS_CLASSES == _mode) {
 			_nextClass = nextAnonymousClass();
 		} else {
-			if ( (result->classLoader == _classLoader) && (NULL != result->arrayClass) ) {
+			if ((result->classLoader == _classLoader) && (NULL != result->arrayClass)) {
 				/* this class is defined in the loader, so follow its array classes */
 				_nextClass = result->arrayClass;
+#if defined(J9VM_OPT_VALHALLA_FLATTENABLE_VALUE_TYPES)
+			} else if ((result->classLoader == _classLoader) && getNullRestrictedArray(result)) {
```

Comment:
`getNullRestrictedArray(result)` here should be `(NULL != getNullRestrictedArray(result))`

Reply:
This is fixed.